src/Microsoft.NetCore.Analyzers/Core/Tasks/DoNotCreateTaskCompletionSourceWithWrongArguments.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Tasks
+{
+    /// <summary>
+    /// RS1027: Do not create TaskCompletionSource with wrong arguments
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotCreateTaskCompletionSourceWithWrongArgumentsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "RS1027";
```

Comment:
Wrong diagnostic ID - RSxxxx diagnostic IDs are reserved for Roslyn.Diagnostics.Analyzers. Please refer to https://github.com/dotnet/roslyn-analyzers/blob/master/GuidelinesForNewRules.md to choose the correct diagnostic ID.

Reply:
@mavasani, that links has:
```
# Microsoft CodeAnalysis API rules
MicrosoftCodeAnalysisCorrectness: RS1000-RS1999
MicrosoftCodeAnalysisDesign: RS1000-RS1999
MicrosoftCodeAnalysisDocumentation: RS1000-RS1999
MicrosoftCodeAnalysisLocalization: RS1000-RS1999
MicrosoftCodeAnalysisPerformance: RS1000-RS1999
MicrosoftCodeAnalysisCompatibility: RS1000-RS1999
```
Is that wrong?

Reply:
Ah sorry, it seems RSxxxx rules used for both `Microsoft.CodeAnalysis.Analyzers` (analyzers for correct usage of public APIs from Microsoft.CodeAnalysis.nupkg, i.e. Roslyn) and `Roslyn.Diagnostic.Analyzers`. All the rules contributed to NetCore/FxCop analyzers should use CAxxxx IDs.

Reply:
Updated to CA2247